### PR TITLE
fixes the gradient checkpointing of whisper

### PR DESCRIPTION
**Pull request by soma2000-lang**
fixes the gradient checkpointing of whisper
@gante 
#21737

### CODE REVIEW

**Review by gante — APPROVED** [review]
The PR looks good, but you'll have to run `make fixup` on your terminal (inside the your `transformers` folder) and push again, to make our CI green :)

**Review by sgugger — APPROVED** [review]
Thanks for your contribution!

File: src/transformers/models/whisper/modeling_whisper.py [hunk 1]
```diff
@@ -1029,6 +1029,12 @@ def forward(
         hidden_states = inputs_embeds + positions
         hidden_states = nn.functional.dropout(hidden_states, p=self.dropout, training=self.training)
 
+        if self.gradient_checkpointing and self.training:
+            if use_cache:
+                logger.warning(
```
Comment (sgugger) [inline src/transformers/models/whisper/modeling_whisper.py:1034]:
```suggestion
                logger.warning(
```
This should be `warn_once` here (or `warning_once`) can't rememver the syntax.

Comment (soma2000-lang) [reply src/transformers/models/whisper/modeling_whisper.py]: @gante done !

File: src/transformers/models/whisper/modeling_whisper.py [hunk 2]
```diff
@@ -1029,6 +1029,12 @@ def forward(
         hidden_states = inputs_embeds + positions
         hidden_states = nn.functional.dropout(hidden_states, p=self.dropout, training=self.training)
 
+        if self.gradient_checkpointing and self.training:
+            if use_cache:
+                logger.warning_once(
+                    "`use_cache = True` is incompatible with gradient checkpointing. Setting `use_cache =" " False`..."
```
Comment (gante) [inline src/transformers/models/whisper/modeling_whisper.py:1035]:
```suggestion
                    "`use_cache = True` is incompatible with gradient checkpointing. Setting `use_cache = False`..."
```

**Review by gante — APPROVED** [review]
One final nit and it is ready to go in :)

File: src/transformers/models/whisper/modeling_whisper.py (hunk 2 above)
Comment (gante) [reply src/transformers/models/whisper/modeling_whisper.py:1035]: (took the liberty of adding it)
